### PR TITLE
Replace columns autocompleter select2 with ng-select

### PR DESCRIPTION
**Pull request by oliverguenther**

### CODE REVIEW

File: frontend/src/app/components/wp-table/configuration-modal/tabs/columns-tab.component.ts [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
 import {ConfigurationService} from 'core-app/modules/common/config/configuration.service';
 import {WorkPackageViewColumnsService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-columns.service';
 import {TabComponent} from 'core-components/wp-table/configuration-modal/tab-portal-outlet';
 import {BannersService} from "core-app/modules/common/enterprise/banners.service";
-
-interface ColumnLike {
-  text:string;
-  id:string;
-}
+import {DraggableOption} from "core-app/modules/common/draggable-autocomplete/draggable-autocomplete.component";
```
Comment (openprojectci) [inline frontend/src/app/components/wp-table/configuration-modal/tabs/columns-tab.component.ts:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/672ca6a93ff52913f26a9f580befb560341731c7/frontend/src/app/components/wp-table/configuration-modal/tabs/columns-tab.component.ts#L9"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>

**Comment by openprojectci** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/672ca6a93ff52913f26a9f580befb560341731c7/frontend/src/app/modules/common/resizer/resizer.component.ts#L20">frontend/src/app/modules/common/resizer/resizer.component.ts#L20</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by lindenthal** [comment]
Hi @oliverguenther,
the long term goal would be to drag and drop the columns in the table view. So instead fixing the current select menu we could make it bigger and replace it entirely. How much effort would this be?

Best
Niels

**Comment by oliverguenther** [comment]
Going to larger route means we first need to be sure how to implement it. I assume you mean dragging the columns for reordering and adding through a dropdown.

We could prioritize this for 10.6. But it is in the order of 1-4 weeks effort.

In any case, I would like to finish this PR to ensure we can remove select2 now. The implementation is done and it requires little to no more effort except for a review.

File: frontend/src/app/components/wp-table/configuration-modal/tabs/columns-tab.component.html [hunk 2]
```diff
@@ -4,6 +4,11 @@
     class="hidden-for-sighted">
   </label>
 
+  <draggable-autocompleter [options]="availableColumnsOptions"
+                           [selected]="selectedColumns"
+                           (onChange)="updateSelected($event)">
+  </draggable-autocompleter>
+
   <ng-container>
     <input #select2Columns
```
Comment (HDinger) [inline frontend/src/app/components/wp-table/configuration-modal/tabs/columns-tab.component.html]: Can't this be removed?

**Review by HDinger — COMMENTED** [review]
Works fine 👍 I have only minor remarks:

* When the list is a bit longer, the columns look a bit off. I think it is okay, to truncate them for now.
Bildschirmfoto 2020-03-26 um 14 16 51

* The class `select2-modal-content` is still applied and can be removed, together with the belonging style rules.

**Comment by HDinger** [comment]
One additional note: I think it would be nice to directly set the focus into the select field again, so that the user can directly start typing when the modal is opened.

File: frontend/src/app/modules/common/draggable-autocomplete/draggable-autocomplete.component.sass [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  display: flex
+  align-items: center
+  justify-content: space-between
+
+  .draggable-autocomplete--remove-item
+    font-size: 8px
+    color: #999
+    padding-left: 5px
+
+.cdk-drag-placeholder
```
Comment (HDinger) [inline frontend/src/app/modules/common/draggable-autocomplete/draggable-autocomplete.component.sass]: I think we don't need these styles any more, right?

**Review by HDinger — APPROVED** [review]
👍 Good to merge once travis is green.

**Pull request merged**